scripts/deployment/test_model_path_detection.py

```
@@ -0,0 +1,114 @@
+        print(f"  Expected: {expected_path}")
+        print(f"  Match: {detected_path == expected_path}")
+        
+        # Clean up environment variable
+        if 'MODEL_BASE_DIR' in os.environ:
+            del os.environ['MODEL_BASE_DIR']
+    
+    # Test 4: With expanduser (~) path
+    print("\n🏠 Test 4: With home directory path expansion")
+    
+    # Create a temporary directory under the expanded home path to ensure it exists
+    with tempfile.TemporaryDirectory() as temp_base:
```

Ruff (0.12.2)

Undefined name `tempfile`

(F821)